validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,177 @@
+	CurrentVersion  string `json:"current_version"`
+	Epoch           string `json:"epoch"`
+}
+
+type RandaoReveal struct {
+	Epoch string `json:"epoch"`
+}
+
+type signResponse struct {
+	Signature string `json:"signature"`
+}
+
+func (client *client) Sign(pubKey string, request *SignRequest) (bls.Signature, error) {
+	requestPath := ethApiNamespace + pubKey
+	jsonRequest, err := json.Marshal(request)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid format, failed to marshal json request")
+	}
+	resp, err := client.doRequest(http.MethodPost, client.BasePath+requestPath, bytes.NewBuffer(jsonRequest))
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body)
+	jsonDataFromHttp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read response body")
+	}
+	signResp := &signResponse{}
+	if err := json.Unmarshal(jsonDataFromHttp, signResp); err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, failed to unmarshal json response")
+	}
+	decoded, err := decodeHex(signResp.Signature)
+	if err != nil {
+		return nil, err
+	}
+	blsSig, err := bls.SignatureFromBytes([]byte(decoded))
+	if err != nil {
+		return nil, err
+	}
+	return blsSig, nil
+}
+
+func (client *client) GetPublicKeys() ([][]byte, error) {
+	const requestPath = "/publicKeys"
+	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp.Body)
+	var publicKeys []string
+	if err := json.NewDecoder(resp.Body).Decode(&publicKeys); err != nil {
+		return nil, errors.Wrap(err, "Invalid Format, unable to read response body as array of strings")
+	}
+	decodedKeys := make([][]byte, len(publicKeys))
+	var errorMessage string
+	for i, value := range publicKeys {
+		decodedKey, err := decodeHex(value)
+		if err != nil {
+			if errorMessage == "" {
+				errorMessage = "Failed to decode from Hex from the following public keys: "
+			}
+			errorMessage += value + " "
+			continue
+		}
+		decodedKeys[i] = decodedKey[:]
+	}
+	if errorMessage != "" {
+		return nil, errors.New(errorMessage)
+	}
+	return decodedKeys, nil
+}
+
+func (client *client) ReloadSignerKeys() error {
+	const requestPath = "/reload"
+	resp, err := client.doRequest(http.MethodPost, client.BasePath+requestPath, nil)
+	if err != nil {
+		return err
+	}
+	defer closeBody(resp.Body)
+	return nil
+}
+
+func (client *client) GetServerStatus() (string, error) {
+	const requestPath = "/upcheck"
+	resp, err := client.doRequest(http.MethodGet, client.BasePath+requestPath, nil)
+	if err != nil {
+		return "", err
+	}
+	defer closeBody(resp.Body)
+	var status string
+	if err := json.NewDecoder(resp.Body).Decode(&status); err != nil {
+		return "", errors.Wrap(err, "Invalid Format, unable to read response body as a string")
+	}
+	return status, nil
+}
+
+func (client *client) doRequest(httpMethod string, fullPath string, body io.Reader) (*http.Response, error) {
```

Comment:
is there a type alias around http method to make this a bit more strict? Maybe http.Method is an alias over string we could use